### PR TITLE
[blocked] CI: Reuse defined workflows

### PR DESCRIPTION
**Pull request by akihironitta**
Blocked by https://github.com/Lightning-AI/devtools/pull/12.

## What does this PR do?
Part of https://github.com/Lightning-AI/lightning/issues/12545.
Partial redo of #832.

### TODO
- [ ] Update azure pipelines file path to use from the azure dashboard.
- [ ] Wait for https://github.com/Lightning-AI/devtools/pull/12, and update the branch name from `ci/update-docs` to `main`.
- [ ] Update all required jobs for merge.

## Before submitting

- [ ] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does **only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? \[not needed for typos/docs\]
- [ ] Did you verify new and **existing tests** pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [ ] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/workflows/pull.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  cancel-in-progress: ${{ ! (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/release/')) }}
+
+jobs:
+  check-code:
+    uses: Lightning-AI/devtools/.github/workflows/check-code.yml@main
+
+  check-schema:
+    uses: Lightning-AI/devtools/.github/workflows/check-schema.yml@main
+    with:
+      azure-dir: ""
```
Comment (Borda) [inline .github/workflows/pull.yml:20]: let's move this config to its own folder
https://github.com/Lightning-AI/lightning-bolts/blob/master/azure-pipelines.yml

Comment (akihironitta) [reply .github/workflows/pull.yml]: Done!

**Review by Borda — COMMENTED** [review]
This rename `azure-pipelines.yml → .azure/gpu-tests.yml` needs to be synced with a manual update on the Azure pipe side...

**Comment by akihironitta** [comment]
Closing this PR in favor of smaller PRs, such as #868.

**Pull request closed without merging**